cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -242,10 +242,15 @@ impl Node<Option<cst::Policy>> {
             .filter_map(|c| {
                 let (e, is_when) = c.to_expr(errs)?;
                 for slot in e.slots() {
-                    errs.push(c.to_ast_err(ToASTErrorKind::SlotsInConditionClause {
-                        slot: (*slot).into(),
-                        clausetype: if is_when { "when" } else { "unless" },
-                    }));
+                    errs.push(ToASTError::new(
+                        ToASTErrorKind::SlotsInConditionClause {
+                            slot: slot.id.into(),
+                            clausetype: if is_when { "when" } else { "unless" },
+                        },
+                        slot.loc.expect(
+                            "all exprs created with `c.to_expr()` should have source locs attached",
```

Comment:
This should require a `PANIC SAFETY` annotation if our CI is working correctly. Also, this invariant feels like it could be easy to accidentally violate in the future. I might prefer falling back to `c.loc`.

Reply:
Good call.